fish-rust/src/ast.rs

```diff
@@ -3192,6 +3192,15 @@ impl<'s> Populator<'s> {
                     }
                 }
             }
+            ParseTokenType::redirection if self.peek_type(0) == ParseTokenType::string => {
+                let next = self.tokens.pop();
```

Comment:
what if the redirection is something like `2>&1`, does that work? We probably don't want to pop the next token in that case

Reply:
ok it's correct, the `1` is actually a separate token

Reply:
I'm glad it turned out not to be an issue, but I hadn't actually considered that.